### PR TITLE
City Centers can no longer be removed by nukes

### PR DESCRIPTION
**Pull request by xlenstra**
Fixes part of #4411

### CODE REVIEW

**Comment by logicminimal** [comment]
:pepehappy:

File: core/src/com/unciv/logic/battle/Battle.kt [hunk 1]
```diff
@@ -618,7 +620,9 @@ object Battle {
         }
 
         // Remove improvements
-        tile.improvement = null
+        if (tile.improvement != null && !attacker.getCivInfo().gameInfo.ruleSet.tileImprovements[tile.improvement]!!.hasUnique("Indestructible")) {
```
Comment (yairm210) [inline core/src/com/unciv/logic/battle/Battle.kt:623]: I'm prettu sure there's a tile.getImprovement() function or whatnot
tileInfo should have a ruleSet reference as well iirc

File: core/src/com/unciv/logic/battle/Battle.kt [hunk 2]
```diff
@@ -558,7 +558,9 @@ object Battle {
         }
 
         // Remove improvements, add fallout
-        tile.improvement = null
+        if (tile.improvement != null && !attacker.getCivInfo().gameInfo.ruleSet.tileImprovements[tile.improvement]!!.hasUnique("Indestructible")) {
```
Comment (yairm210) [inline core/src/com/unciv/logic/battle/Battle.kt:561]: Here too, same as below, !attacker.getTile().getTileImprovement()!!.hasUnique("Indestructible")

Comment (xlenstra) [reply core/src/com/unciv/logic/battle/Battle.kt]: I probably should have thought of that, should be fixed now.

**Comment by logicminimal** [comment]
this pr doesn't make nukes unable to destroy razeable cities does it?

**Comment by xlenstra** [comment]
No, because if a city would be destroyed, the function city.destroy() is called. This PR only makes it so the nuke won't remove City Center improvements

**Pull request merged**